File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webapi/service/types/TranslateContext.java
package org.opcfoundation.webapi.service.types;

import org.eclipse.milo.opcua.stack.core.types.structured.BrowsePath;
import org.eclipse.milo.opcua.stack.core.types.structured.RequestHeader;
import org.springframework.lang.Nullable;
import org.springframework.web.context.request.NativeWebRequest;

import java.util.List;

public class TranslateContext extends ServiceContext {
    private final List<BrowsePath> nodesToTranslate;

    public TranslateContext(
            List<BrowsePath> nodesToTranslate,
            @Nullable String serverUri,
            @Nullable RequestHeader header,
            @Nullable NativeWebRequest webRequest)
    {
        super(serverUri,header,webRequest);
        this.nodesToTranslate = nodesToTranslate;
    }

    public List<BrowsePath> getNodesToTranslate() {
        return nodesToTranslate;
    }
}

File: opcua-webservice-node/src/server/types/contexts/TranslateContext.ts
import { BrowsePath, TranslateBrowsePathsToNodeIdsRequest } from "opcua-webapi";
import { ServiceContext } from "./ServiceContext";

export class TranslateContext extends ServiceContext
{
    private _browsePaths: Array<BrowsePath>;
    
    constructor(request: TranslateBrowsePathsToNodeIdsRequest, serverUri?: string)
    {
        super(serverUri, request.RequestHeader);
        this._browsePaths = (request.BrowsePaths) ? request.BrowsePaths : [];
    }

    get browsePaths() : Array<BrowsePath>
    {
        return this._browsePaths;
    }
}